app-text/zeal/files/qt6.patch

```diff
@@ -0,0 +1,11 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -22,7 +22,7 @@
+ set(PROJECT_COPYRIGHT "© 2013-2024 Oleg Shparber and other contributors")
+ 
+ # Find available major Qt version. It will be stored in QT_VERSION_MAJOR.
+-find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core)
++find_package(QT NAMES Qt6 REQUIRED COMPONENTS Core)

```

Comment:
In theory this should never matter since Qt6 would always be found -- it's a dependency after all -- so it shouldn't be able to fall back to Qt5. Any idea why it did so anyway? Did the configure logs suggest that it searched and didn't find Qt6?

Reply:
The default logs didn't mention anything about this or any specifics on how cmake picks Qt version or what version it picks. Maybe there's some log that'd be more verbose than the default configure log, but I haven't checked any.
<details><summary>Log segment</summary>

```
>>> Preparing source in /tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1 ...                                                                                                                                                               
 * Source directory (CMAKE_USE_DIR): "/tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1"                                                                                                                                                    
 * Build directory  (BUILD_DIR):     "/tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1_build"                                                                                                                                              
 * Applying disable-werror.patch ...                                                                                                                                                                                                [ ok ] 
>>> Source prepared.                                                                                                                                                                                                                       
>>> Configuring source in /tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1 ...                                                                                                                                                             
 * Source directory (CMAKE_USE_DIR): "/tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1"                                                                                                                                                    
 * Build directory  (BUILD_DIR):     "/tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1_build"                                                                                                                                              
cmake -C /tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DZEAL_RELEASE_BUILD=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=/tmp/portage/app-text/zeal
-0.7.1/work/zeal-0.7.1_build/gentoo_toolchain.cmake /tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1                                                                                                                                       
loading initial cache file /tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1_build/gentoo_common_config.cmake                                                                                                                               
-- The CXX compiler identification is GNU 14.1.1                                                                                                                                                                                           
-- Detecting CXX compiler ABI info                                                                                                                                                                                                         
-- Detecting CXX compiler ABI info - done                                                                                                                                                                                                  
-- Check for working CXX compiler: /usr/bin/x86_64-pc-linux-gnu-g++ - skipped                                                                                                                                                              
-- Detecting CXX compile features                                                                                                                                                                                                          
-- Detecting CXX compile features - done                                                                                                                                                                                                   
Calculated Zeal version: 0.7.1                                                                                                                                                                                                             
-- Installing in the same prefix as Qt, adopting their path scheme.                                                                                                                                                                        
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD                                                                                                                                                                                                 
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success                                                                                                                                                                                       
-- Found Threads: TRUE                                                                                                                                                                                                                     
-- Found X11: /usr/include                                                                                                                                                                                                                 
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so                                                                                                                                                                  
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so - found                                                                                                                                                          
-- Looking for gethostbyname                                                                                                                                                                                                               
-- Looking for gethostbyname - found                                                                                                                                                                                                       
-- Looking for connect                                                                                                                                                                                                                     
-- Looking for connect - found                                                                                                                                                                                                             
-- Looking for remove                                                                                                                                                                                                                      
-- Looking for remove - found                                                                                                                                                                                                              
-- Looking for shmat                                                                                                                                                                                                                       
-- Looking for shmat - found                                                                                                                                                                                                               
-- Looking for IceConnectionNumber in ICE                                                                                                                                                                                                  
-- Looking for IceConnectionNumber in ICE - found                                                                                                                                                                                          
-- Found XCB_XCB: /usr/lib64/libxcb.so (found version "1.17.0")                                                                                                                                                                            
-- Found XCB_KEYSYMS: /usr/lib64/libxcb-keysyms.so (found version "0.4.1")                                                                                                                                                                 
-- Found XCB: /usr/lib64/libxcb.so;/usr/lib64/libxcb-keysyms.so (found version "1.17.0") found components: XCB KEYSYMS                                                                                                                     
-- Found SQLite3: /usr/include (found version "3.46.0")                                                                                                                                                                                    
-- Project executable name: zeal                                                                                                                                                                                                           
-- <<< Gentoo configuration >>>                        
Build type      RelWithDebInfo                                                                                                                                                                                                             
Install path    /usr
Compiler flags:
C               
C++             -O2 -pipe -march=alderlake -mabm -mno-cldemote -mno-kl -mno-pconfig -mno-sgx -mno-widekl -mshstk --param=l1-cache-line-size=64 --param=l1-cache-size=32 --param=l2-cache-size=24576
Linker flags:
Executable      -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
Module          -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
Shared          -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs

-- Configuring done (2.2s)
-- Generating done (0.2s)
-- Build files have been written to: /tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1_build
>>> Source configured.
>>> Compiling source in /tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1 ...
 * Source directory (CMAKE_USE_DIR): "/tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1"
 * Build directory  (BUILD_DIR):     "/tmp/portage/app-text/zeal-0.7.1/work/zeal-0.7.1_build"
ninja -v -j20 -l20
```

</details>

Reply:
Whatever hat does in theory, in practise we can't rely on this automatic cmake stupidity.

Reply:
Well, I'd be interested in hearing if this is something that can be fixed upstream then.

Since as you say it's stupid and doesn't work and isn't even correct cmake in the first place.

Reply:
I've asked upstream in [a bug report](https://github.com/zealdocs/zeal/issues/1647) whether an explicit way to select Qt5/Qt6 could be exposed. Would a patch-based Qt6-only version be acceptable or do I need to implement Qt version switching?

Reply:
In other ebuilds I've used sed, patch is ofc always cleaner.